### ISSUE TITLE
Modify the usage of NewSecretClient to reflect the recent changes on go-mod-secrets

### ISSUE THREAD

**Issue by jim-wang-intel** [edgexfoundry/go-mod-bootstrap#35]
The constructor or instantiator of `vault.NewSecretclient` from go-mod-secrets client is changed after [the Pull Request of token-refresh background process](https://github.com/edgexfoundry/go-mod-secrets/pull/45)  has been merged onto master branch.  

This requires code modification on the client or caller side.  In particular, the `NewSecretClient` is now instantiated with a `NewSecretClientFactory` object first: 
```go
NewSecretClientFactory()
```
 and `NewSecretClient` takes extra new input arguments with background `context`, referential implementation of `loggingClient` and `tokenExpiredCallback` func:
```go
func (factory *secretClientFactory) NewSecretClient(ctx context.Context,
	config SecretConfig,
	lc loggingClient,
	tokenExpiredCallback tokenExpiredCallback) (pkg.SecretClient, error) { ... }
```
With new secret client factory, the caller now is able to instantiate multiple secret clients for use cases like dynamic secrets on the fly with potentially asynchronous fashion.

The current code reference to be modified in go-mod-bootstrap is here -- https://github.com/edgexfoundry/go-mod-bootstrap/blob/fe266a717da291b9ad28584cefff274bcef081c0/bootstrap/handlers/secret/secret.go#L64. 

## Suggested solution
To handle the token expired scenario, implement the default callback to re-read the token from the disk to see if it was updated offline and set the retry to be true while supplying a new token if any.
Some code reference is here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/master/bootstrap/handlers/secret/secret.go#L100-L108  and use that similar logic to compare the new token to the old one with retry logic for the callback function.

## Dependency
Dependent on edgexfoundry/go-mod-secrets#34

**Comment by jim-wang-intel** [comment]
cc: @michaelestrin @bnevis-i  @lenny-intel  as I do not have permission to edit on this issue to add any label or milestone, please do it accordingly for the planning purposes.

**Comment by michaelestrin** [comment]
@jim-wang-intel Updated issue with labels.  Assigned you.  Please advise if you are not currently working on this (or plan on doing so); @AnthonyMBonafide can pick this work up if you haven't already.

**Comment by bnevis-i** [comment]
Closed by https://github.com/edgexfoundry/go-mod-bootstrap/pull/43

**Comment by jim-wang-intel** [comment]
This issue is closed by PR #43 merged into master already.
